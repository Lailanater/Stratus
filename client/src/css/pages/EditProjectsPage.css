#editProjectsPageButton {
  margin-top: 20px;
  margin-bottom: 25px;
}
