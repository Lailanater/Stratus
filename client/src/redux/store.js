import { createStore } from 'redux';
import rootReducer from './reducers';
import { loadState } from '../localStorage';

let initialState = loadState();

if (initialState === undefined) {
  initialState = {
    isAppMenuDisplayed: false,
    theme: 'dark',
    currentProject: '',
    projects: [],
    menuName: '',
    needsRepeat: false,
    needsDtmf: false,
    needsVoice: false,
    dtmfOptions: [],
    defaultRouteTo: 'menu',
    isNextButtonDisabled: false,
    isFinishButtonDisabled: false
  };
}

export default createStore(rootReducer, initialState);
