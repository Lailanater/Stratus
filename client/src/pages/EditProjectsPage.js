import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import ProjectList from '../components/ProjectList';
import { Redirect } from 'react-router';
import '../css/pages/EditProjectsPage.css';

const EditProjectsPage = () => {
  const [canRedirect, setCanRedirect] = useState(false);

  if (canRedirect) {
    return <Redirect to="/addProject" />;
  }

  return (
    <div className="centerText">
      <Button
        id="editProjectsPageButton"
        style={{ backgroundColor: '#4caf50' }}
        onClick={() => setCanRedirect(true)}
      >
        Add Project
      </Button>
      <ProjectList />
    </div>
  );
};

export default EditProjectsPage;
