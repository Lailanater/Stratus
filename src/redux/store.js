import {createStore} from "redux";
import rootReducer from "./reducers";

export const initialState = {
    isAppMenuDisplayed: false,
    theme: "dark",
    currentProject: "",
    projects: []
};

export default createStore(rootReducer);