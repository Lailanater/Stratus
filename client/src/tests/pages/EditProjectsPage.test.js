import React from 'react';
import renderer from 'react-test-renderer';
import EditProjectsPage from '../../pages/EditProjectsPage';
import store from '../../redux/store';
import { Provider } from 'react-redux';

it('renders without crashing', () => {
  const component = renderer.create(
    <Provider store={store}>
      <EditProjectsPage />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
