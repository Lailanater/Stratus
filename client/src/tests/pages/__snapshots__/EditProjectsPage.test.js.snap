// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="centerText"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#4caf50",
        "marginBottom": "25px",
        "marginTop": "20px",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Add Project
    </span>
  </button>
  <div
    id="fileExistsDialogButtonContainer"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
    >
      You have no projects yet. Try adding one!
    </h4>
  </div>
</div>
`;
